### ISSUE TITLE
twisted.mail.smtp.xtext_encode and xtext_decode take the wrong number of arguments

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2976]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2976|
|Type|defect|
|Created|2008-01-05 15:07:02Z|
|Branch|https://github.com/twisted/twisted/tree/xtext-signature-2976|

See [#2929](https://github.com/twisted/twisted/issues/2929)

<details><summary>Searchable metadata</summary>

```
trac-id__2976 2976
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_xtext_signature_2976 branches/xtext-signature-2976
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1199545622000000 1199545622000000
changetime__1222108604000000 1222108604000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [24716]) Branching to 'xtext-signature-2976'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @exarkun|
|-|-|

 * The fix in xtext_decode isn't there if _c_urlarg is present
 * The test fails under python2.3:
exceptions.TypeError: encoder did not return a string object (type=unicode)
 * there is a typo in xtext_decode: erorrs

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [24872]) test fast and slow versions and fix spelling of `errors`

refs [#2976](https://github.com/twisted/twisted/issues/2976)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun removed owner|
|-|-|

Thanks!  Should all be fixed now.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set owner to @exarkun|
|-|-|

The import "from email.base64MIME import encode as encode_base64" seems to work on all python version. Do you think we can remove it?

Please merge when ready.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

(In [24904]) Remove Python 2.2 support code

refs [#2976](https://github.com/twisted/twisted/issues/2976)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [24905]) Merge xtext-signature-2976

Author: exarkun
Reviewer: therve
Fixes: [#2976](https://github.com/twisted/twisted/issues/2976)

Change the xtext encoder and decoder functions defined by
`twisted.mail.smtp` so that they accept the `errors`
parameter.  The parameter is ignored, but it must be accepted
or callers of `str.decode` and `unicode.encode` which pass
a value will be thrown a `TypeError`.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
